### DISCUSSION TITLE
call of overloaded ‘bind is ambiguous  ( Poco::Data::Keywords::useRef + Poco::Nullable<bool> + const_iterator)

### DISCUSSION THREAD

**Discussion by omerbrandis** [pocoproject/poco#4436] [Q&A]
I'm using a fairly recent version of Poco , from 7.2023

getting the error 
/home/omerbrandis/Downloads/poco-devel-24.7.2023/Data/include/Poco/Data/TypeHandler.h:121: error: call of overloaded ‘bind(std::size_t&, const Poco::Nullable<bool>&, Poco::Data::AbstractBinder::Direction&)’ is ambiguous
   pBinder->bind(pos, obj, dir);
   ^~~~~~~

My code 
```
for (std::vector<myStructure>::const_iterator It = myVector.begin() ; It != myVector.end(); It++)
{
   .
   .
   SQLStmt << InsertString
   .
   .
  , Poco::Data::Keywords::useRef(It->Struct.member)
}
```

however using a temporary works:
```
Poco::Nullable<bool> Temp  = It->Struct.member;
, Poco::Data::Keywords::useRef(Temp)

```

Can someone point me in the right direction?

**Comment by andrewauclair** [comment]
What compiler and version? These types of issues are virtually impossible to help with if the compiler isn't known.

**Comment by omerbrandis** [comment]
gcc (Ubuntu 8.4.0-1ubuntu1~16.04.1) 8.4.0

**Comment by matejk** [comment]
Which version of Poco do you try to compile?

Does it work with latest version of the code?

FYI: Latest Poco (1.13) requires a compiler that supports C++17. GCC 8 does not fully support that standard.

**Comment by aleks-f** [comment]
[SSCCE](http://sscce.org) please

**Comment by omerbrandis** [comment]
Hello guys, 
thank you for trying to help.

I think I'm running poco version 1.12.3 ( as is written in the "VERSION" file in the root dir of the poco source)

attached is an SSCCE.

[sscce.txt](https://github.com/pocoproject/poco/files/14176383/sscce.txt)

:-)
